### PR TITLE
fix parsing of CDI yaml

### PR DESCRIPTION
**Pull request by mikebonnet**
A previous commit removed pyyaml as a dependency and replaced it with code which parses the yaml line-by-line looking for "name" and using the following token as the device name. However, the code did not handle the hyphen prefix used in yaml lists, so it was ignoring all entries in the yaml, preventing use of "cuda" acceleration. Fix the parser to correctly handle hyphens.

## Summary by Sourcery

Fix load_cdi_yaml to correctly handle hyphen-prefixed list entries and expand coverage with unit tests for YAML parsing, configuration loading, and device lookup

Bug Fixes:
- Handle "- name" keys when parsing CDI YAML lists to include all device entries

Tests:
- Add parameterized tests for load_cdi_yaml covering empty lists, numeric, "all", and UUID names
- Add tests for load_cdi_config to verify configuration file discovery and parsing
- Add tests for find_in_cdi to validate visibility filtering and no-config fallback

### CODE REVIEW

File: test/unit/test_common.py [hunk 1]
```diff
@@ -183,3 +193,58 @@ def test_default_get_accel(self):  # sourcery skip: no-loop-in-tests
...
+    [
+        (StringIO(CDI_NODEV), {"devices": []}),
+        (StringIO(CDI_ZERO), {"devices": [{"name": "0"}]}),
+        (StringIO(CDI_ALL), {"devices": [{"name": "all"}]}),
+        (StringIO(CDI_UUID), {"devices": [{"name": CDI_GPU_UUID}]}),
+        (StringIO(CDI_EVERYTHING), {"devices": [{"name": CDI_GPU_UUID}, {"name": "0"}, {"name": "all"}]}),
+    ],
+)
+def test_load_cdi_yaml(input: str, result: CDI_RETURN_TYPE):
+    assert load_cdi_yaml(input) == result
```
[the remark this answers is not in the document]
Comment (mikebonnet) [reply test/unit/test_common.py:221]: Unnecessary.

File: test/unit/test_common.py [hunk 2]
```diff
@@ -183,3 +193,58 @@
...
+
+
+@patch("os.walk", return_value=(("/etc/cdi", None, ("nvidia.yaml",)),))
+@patch("builtins.open", mock_open(read_data=CDI_EVERYTHING))
+def test_load_cdi_config(mock_open):
```
[the remark this answers is not in the document]
Comment (mikebonnet) [reply test/unit/test_common.py]: Unnecessary.

File: test/unit/test_common.py [hunk 3]
```diff
@@ -183,3 +193,58 @@
...
+CDI_GPU_UUID = "GPU-abcdefgh"
+CDI_NODEV = """---
+cdiVersion: 0.5.0
+kind: nvidia.com/gpu
+devices:
+"""
+CDI_ZERO = CDI_NODEV + '  - name: "0"\n'
+CDI_ALL = CDI_NODEV + "  - name: all\n"
+CDI_UUID = CDI_NODEV + f"  - name: {CDI_GPU_UUID}\n"
+CDI_EVERYTHING = CDI_UUID + '  - name: "0"\n  - name: all\n'
```
[the remark this answers is not in the document]
Comment (mikebonnet) [reply test/unit/test_common.py:207]: Unnecessary.

File: ramalama/common.py [hunk 4]
```diff
@@ -251,7 +251,7 @@ def load_cdi_yaml(stream: Iterable[str]) -> CDI_RETURN_TYPE:
     for line in stream:
         if ':' in line:
             key, value = line.split(':', 1)
-            if key.strip() == "name":
+            if key.strip() == "- name":
```
[the remark this answers is not in the document]
Comment (mikebonnet) [reply ramalama/common.py:254]: Unnecessary.

File: test/unit/test_common.py [hunk 5]
```diff
@@ -183,3 +193,58 @@ def test_default_get_accel(self):  # sourcery skip: no-loop-in-tests
...
+
+@pytest.mark.parametrize(
+    "input,result",
+    [
+        (StringIO(CDI_NODEV), {"devices": []}),
+        (StringIO(CDI_ZERO), {"devices": [{"name": "0"}]}),
+        (StringIO(CDI_ALL), {"devices": [{"name": "all"}]}),
+        (StringIO(CDI_UUID), {"devices": [{"name": CDI_GPU_UUID}]}),
+        (StringIO(CDI_EVERYTHING), {"devices": [{"name": CDI_GPU_UUID}, {"name": "0"}, {"name": "all"}]}),
+    ],
```
[the remark this answers is not in the document]
Comment (mikebonnet) [reply test/unit/test_common.py:218]: Unnecessary.

**Review by engelmi — APPROVED** [review]
LGTM
